app/src/main/java/com/kodeco/android/countryinfo/MainActivity.kt

```diff
@@ -32,7 +33,17 @@ class MainActivity : ComponentActivity() {
                     modifier = Modifier.fillMaxSize(),
                     color = Color.White
                 ) {
-                    ApplicationNavigation(repository)
+                    var isAnimationFinished by rememberSaveable { mutableStateOf(false) }
+
+                    LaunchedEffect(key1 = Unit) {
+                        delay(3000)
+                        isAnimationFinished = true
+                    }
+
+                    when (isAnimationFinished) {
+                        false -> SplashScreen()
+                        true -> ApplicationNavigation(repository)
+                    }
```

Comment:
Nice job here! As mentioned you could also do this directly in the ApplicationNavigation where the navhost lives.

```kotlin
            composable(route = Screens.Splash.path) {
                LaunchedEffect(null) {
                    delay(3000)
                    navController.navigate(Screen.CountryList.route){
                        popUpTo(Screen.Splash.route){
                            inclusive = true
                        }
                        launchSingleTop = true
                    }
                }
                SplashScreen()
```

Reply:
thanks, this helped me a lot!